crates/engine/tree/src/tree/root.rs

```diff
@@ -41,20 +58,45 @@ impl<Factory> StateRootTask<Factory> {
         Self { consistent_view, state_stream, input }
     }
 
+    /// Spawns the state root task and returns a handle to await its result.
+    pub(crate) fn spawn(self) -> StateRootHandle {
+        let (tx, rx) = mpsc::channel();
+
+        // Spawn the task that will process state updates and calculate the root
+        std::thread::Builder::new()
+            .name("State Root Task".to_string())
+            .spawn(move || {
+                debug!(target: "engine::tree", "Starting state root task");
+                let result = self.run();
+                let _ = tx.send(result);
+            })
+            .expect("failed to spawn state root thread");
+        StateRootHandle { rx }
+    }
+
     /// Handles state updates.
-    pub(crate) fn on_state_update(&self, _update: EvmState) {
+    fn on_state_update(
+        _view: &ConsistentDbView<Factory>,
+        _input: &Arc<TrieInput>,
+        _state: EvmState,
+    ) {
         // TODO: calculate hashed state update and dispatch proof gathering for it.
     }
-}
 
-impl<Factory> Future for StateRootTask<Factory> {
-    type Output = Result<(B256, TrieUpdates), ParallelStateRootError>;
+    fn run(self) -> StateRootResult {
+        let Self { state_stream, consistent_view, input } = self;
+
+        let mut receiver = state_stream.into_inner();
+
+        // Process all items until the channel is closed
+        while let Ok(state) = receiver.try_recv() {
+            Self::on_state_update(&consistent_view, &input, state);
+        }
 
-    fn poll(self: Pin<&mut Self>, _cx: &mut Context<'_>) -> Poll<Self::Output> {
         // TODO:
-        //    * poll incoming state updates stream
         //    * keep track of proof calculation
         //    * keep track of intermediate root computation
-        Poll::Pending
+        //    * return final state root result
+        Ok((B256::default(), TrieUpdates::default()))
     }
```

Comment:
we'll do a bunch of polling & spawning here, i'd prefer to keep this a future

Reply:
sre done here https://github.com/paradigmxyz/reth/pull/12335/commits/53582dd340edbe4118a76ad74045e5bea890f117 PTAL